include/ensmallen_bits/sgd/update_policies/weight_decay_momentum_update.hpp

```diff
@@ -0,0 +1,101 @@
+/**
+ * @file weight_decay_momentum_update.hpp
+ * @author Ajinkya Tejankar
+ * @author Niteya Shah
+ * Momentum update with weight decay for Stochastic Gradient Descent
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_WEIGHT_DECAY_MOMENTUM_UPDATE_HPP
+#define ENSMALLEN_WEIGHT_DECAY_MOMENTUM_UPDATE_HPP
+
+namespace ens {
+
+/**
+ * De-Coupled Weight Decay Policy for SGD with Momentum (SGDW).
+ *
+ * This Implments Decoupled Weight Decay Policy  in which the weight decay is
+ * decoupled from the optimization steps w.r.t. to the loss function.
+ * For SGD variants, this simplifies hyperparameter search since it decouples
+ * the settings of weight decay and learning rate.
+ *
+ *
+ * The Update Policy for SGDR
+ * \f[
+ * v = mu*v + \alpha \nabla f_i(A)
+ * A_{j + 1} = A_j - \nabla \lamdba A_j - v
+ * \f]
+ *
+ * @code
+ * @article{
+ *   title   = {Decoupled Weight Decay Regularization},
+ *   author  = {{Ilya}, L. and {Frank}, H.},
+ *   journal = {ArXiv e-prints},
+ *   url     = {https://arxiv.org/pdf/1711.05101.pdf}
+ *   year    = {2019}
+ * }
```

Comment:
Please fix this, and please add a comment above the `@code` block pointing out that this is the reference to the paper where this strategy is defined.